include/graph.h

```diff
@@ -1,23 +1,37 @@
-#pragma once
+#ifndef MAIN_GRAPH_H
+#define MAIN_GRAPH_H
 #include <cstdlib>
 #include <exception>
 #include <set>
+#include <fstream>
 #include "supernode.h"
 
+#ifdef VERIFY_SAMPLES_F
+#include "../test/util/deterministic.h"
+#endif
+
 enum UpdateType {
-  INSERT,
-  DELETE,
+  INSERT = 0,
+  DELETE = 1,
 };
 
 typedef pair<Edge, UpdateType> GraphUpdate;
 
+/**
+ * Undirected graph object with n nodes labelled 0 to n-1, no self-edges,
+ * multiple edges, or weights.
+ */
 class Graph{
   const uint64_t num_nodes;
   bool update_locked = false;
   // a set containing one "representative" from each supernode
   set<Node>* representatives;
   Supernode** supernodes;
 
+#ifdef VERIFY_SAMPLES_F
+  std::string cum_in = "./cum_sample.txt";

```

Comment:
I think this should be moved to the ifdef on line 49. As far as I can see cum_in isn't used before that point. Fewer ifdefs = simpler.